sycl/include/CL/sycl/aliases.hpp

```diff
@@ -0,0 +1,115 @@
+class half;
+} // namespace half_impl
+} // namespace detail
+} // namespace sycl
+} // namespace cl
+
+#ifdef __SYCL_DEVICE_ONLY__
+using half = _Float16;
+#else
+using half = cl::sycl::detail::half_impl::half;
+#endif
+
+#define MAKE_VECTOR_ALIAS(ALIAS, TYPE, N)                                      \
+  using ALIAS##N = cl::sycl::vec<TYPE, N>;
+
+#define MAKE_VECTOR_ALIASES_FOR_ARITHMETIC_TYPES(N)                            \
+  MAKE_VECTOR_ALIAS(char, char, N)                                             \
+  MAKE_VECTOR_ALIAS(short, short, N)                                           \
+  MAKE_VECTOR_ALIAS(int, int, N)                                               \
+  MAKE_VECTOR_ALIAS(long, long, N)                                             \
+  MAKE_VECTOR_ALIAS(float, float, N)                                           \
+  MAKE_VECTOR_ALIAS(double, double, N)                                         \
+  MAKE_VECTOR_ALIAS(half, ::half, N)
+
+#define MAKE_VECTOR_ALIASES_FOR_OPENCL_TYPES(N)                                \
+  MAKE_VECTOR_ALIAS(cl_char, cl::sycl::cl_char, N)                             \
+  MAKE_VECTOR_ALIAS(cl_uchar, cl::sycl::cl_uchar, N)                           \
+  MAKE_VECTOR_ALIAS(cl_short, cl::sycl::cl_short, N)                           \
+  MAKE_VECTOR_ALIAS(cl_ushort, cl::sycl::cl_ushort, N)                         \
+  MAKE_VECTOR_ALIAS(cl_int, cl::sycl::cl_int, N)                               \
+  MAKE_VECTOR_ALIAS(cl_uint, cl::sycl::cl_uint, N)                             \
+  MAKE_VECTOR_ALIAS(cl_long, cl::sycl::cl_long, N)                             \
+  MAKE_VECTOR_ALIAS(cl_ulong, cl::sycl::cl_ulong, N)                           \
+  MAKE_VECTOR_ALIAS(cl_float, cl::sycl::cl_float, N)                           \
+  MAKE_VECTOR_ALIAS(cl_double, cl::sycl::cl_double, N)                         \
+  MAKE_VECTOR_ALIAS(cl_half, cl::sycl::half, N)
+
+#define MAKE_VECTOR_ALIASES_FOR_SIGNED_AND_UNSIGNED_TYPES(N)                   \
+  MAKE_VECTOR_ALIAS(schar, signed char, N)                                     \
+  MAKE_VECTOR_ALIAS(uchar, unsigned char, N)                                   \
+  MAKE_VECTOR_ALIAS(ushort, unsigned short, N)                                 \
+  MAKE_VECTOR_ALIAS(uint, unsigned int, N)                                     \
+  MAKE_VECTOR_ALIAS(ulong, unsigned long, N)                                   \
+  MAKE_VECTOR_ALIAS(longlong, long long, N)                                    \
+  MAKE_VECTOR_ALIAS(ulonglong, unsigned long long, N)
+
+#define MAKE_VECTOR_ALIASES_FOR_VECTOR_LENGTH(N)                               \
+  MAKE_VECTOR_ALIASES_FOR_ARITHMETIC_TYPES(N)                                  \
+  MAKE_VECTOR_ALIASES_FOR_OPENCL_TYPES(N)                                      \
+  MAKE_VECTOR_ALIASES_FOR_SIGNED_AND_UNSIGNED_TYPES(N)
+
+namespace cl {
+namespace sycl {
+namespace detail {
+using conditional_data_t = decltype(1 == 1);
+} // namespace detail
+using byte = ::uint8_t;
```

Comment:
Why not `std::uint8_t` and other `std::` below?

Reply:
Added.